scripts/multi_component_hydrogen_bond_propensity/multi_component_hydrogen_bond_propensity_report.py

```diff
@@ -343,22 +345,35 @@ def main(structure, work_directory, library, csdrefcode):
         crystal = crystal_reader[0]
 
         directory = os.path.join(os.path.abspath(work_directory), crystal.identifier)
-
-        try:
-            propensities, donors, acceptors = propensity_calc(crystal, directory)
-            coordination_scores = coordination_scores_calc(crystal, directory)
-            pair_output(crystal.identifier, propensities, donors, acceptors, coordination_scores, directory)
-            mc_dictionary[coformer_name] = get_mc_scores(propensities, crystal.identifier)
-
-        except RuntimeError:
-            print("Propensity calculation failure for %s!" % coformer_name)
-            mc_dictionary[coformer_name] = ["N/A", "N/A", "N/A", "N/A", "N/A", crystal.identifier]
+        if os.path.exists(os.path.join(directory, "success.json")):
+            with open(os.path.join(directory, "success.json"), "r") as file:
+                tloaded = json.load(file)
+            mc_dictionary[coformer_name] = tloaded
+        else:
+            try:
+                propensities, donors, acceptors = propensity_calc(crystal, directory)
+                coordination_scores = coordination_scores_calc(crystal, directory)
+                pair_output(crystal.identifier, propensities, donors, acceptors, coordination_scores, directory)
+                with open(os.path.join(directory, "success.json"), "w") as file:
+                    tdata = get_mc_scores(propensities, crystal.identifier)
+                    json.dump(tdata, file)
+                mc_dictionary[coformer_name] = get_mc_scores(propensities, crystal.identifier)
+                print(get_mc_scores(propensities, crystal.identifier))
+            except RuntimeError:
+                print("Propensity calculation failure for %s!" % coformer_name)
+                mc_dictionary[coformer_name] = ["N/A", "N/A", "N/A", "N/A", "N/A", crystal.identifier]
+                failures.append(coformer_name)
 
     # Make sense of the outputs of all the calculations
     mc_hbp_screen = sorted(mc_dictionary.items(), key=lambda e: 0 if e[1][0] == 'N/A' else e[1][0], reverse=True)
     diagram_file = make_diagram(api_molecule, work_directory)
     chart_file = make_mc_chart(mc_hbp_screen, directory, api_molecule)
     make_mc_report(structure, mc_hbp_screen, work_directory, diagram_file, chart_file)
+    with open("failures.txt", 'w') as file:
```

Comment:
I think this should have the same output location as the above with the 
```python
os.path.join(directory,...) 
```

Reply:
I have libraries of coformers with 90+ elements. Looking at each one individually to verify if it failed would be very tiresome. That's why I did it outside the directory of each coformer. That file lists the failures and then I can look inside the failed dir.